src/services/Storage.ts

```diff
@@ -17,3 +24,92 @@ export const createNote = async (
   };
   return newPostFeedItem;
 };
+
+export class Store implements StoreInterface {
+  put(targetPath: string, _content: Content): Promise<URL> {
+    return Promise.resolve(
+      new URL(`${process.env.REACT_APP_UPLOAD_HOST}/${targetPath}`)
+    );
+  }
+
+  async putStream(
+    targetPath: string,
+    doWriteToStream: WriteStreamCallback
+  ): Promise<URL> {
+    const ws = new ServerWriteStream(targetPath);
+    await doWriteToStream(ws);
+    return new URL(`${process.env.REACT_APP_UPLOAD_HOST}/${targetPath}`);
+  }
+}
+
+type WriteCallback = (error: Error | null | undefined) => void;
+type EndCallback = () => void;
+
+/* eslint-disable no-dupe-class-members */
+class ServerWriteStream implements WriteStream {
+  chunks: Uint8Array[] = [];
+  targetPath: string;
+
+  constructor(path: string) {
+    this.targetPath = path;
+  }
+
+  write(chunk: unknown, encoding?: string, callback?: WriteCallback): boolean;
+  write(chunk: unknown, cb?: WriteCallback): boolean;
+
+  write(
+    chunk: unknown,
+    param1?: string | WriteCallback,
+    param2?: WriteCallback
```

Comment:
you might also be able clean this up with a splat.

`write(chunk: unknown, ...args)`

Not sure with the types though